File: Hashmap/p242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashS=defaultdict(int)
        hashT= defaultdict(int)

        for c in s:
            hashS[c]+=1
        
        for c in t:
            hashT[c]+=1
        
        if hashT==hashS:
            return True
        
        return False
    
#time complexity O(N + M)
#space complexity O(N+M)

File: Hashmap/p242_valid_anagram.cpp
class Solution {
public:
  bool isAnagram(string s, string t) {
    unordered_map<char, int> mapS;
    unordered_map<char, int> mapT;

    for (const auto &c : s) {
      mapS[c] += 1;
    }

    for (const auto &c : t) {
      mapT[c] += 1;
    }

    if (mapS == mapT) {
      return true;
    }
    return false;
  }
};

// time complexity O(N)
// space complexity O(N)